shared/chat/conversation/messages/wrapper/chat-send.js

```diff
@@ -76,19 +69,40 @@ class SendIndicator extends React.Component<Props, State> {
     } else if (this.props.failed) {
       // previously failed message
       this._onFailed()
+      this._setStatus('error')
     } else if (this.props.sent) {
       // previously sent message
       this._setVisible(false)
     }
   }
 
-  componentWillReceiveProps(nextProps: Props) {
-    if (nextProps.failed && !this.props.failed) {
+  static getDerivedStateFromProps(props, state) {
```

Comment:
is there a reason why this can't just use componentDidUpdate and not getDerivedStateFromProps?

Reply:
No, I thought that was the upgrade path recommended in the article.